packages/docs/docgen.js

```diff
@@ -8,16 +8,25 @@ const createJsdocContent = require('./functions-generator');
 
 const COMPONENTS_JSON = 'component-docs.json';
 
+const DEFAULT_TS_BABEL_OPTIONS  = {
+    configFile: './.docgen.babelrc',
+    root: __dirname
+};
+
 const componentsSrc = path.resolve(__dirname, '../*/src');
-const files = glob.sync(`${componentsSrc}/**/*.js`).filter(file => !file.match(/((test|spec|index).js|(\/__mock__\/|\/__mocks__\/|\/test\/))/gmi));
+const files =
+    glob
+    .sync(`${componentsSrc}/**/*.{js,ts,tsx}`)
```

Comment:
How about excluding `d.ts`?
```suggestion
    .sync(`${componentsSrc}/**/*.[!d.]{js,ts,tsx}`)
```

Not sure about the dot, if it needs to be escaped or not.

Reply:
@karelhala the `.d.ts` files are filtered on the line below. The glob pattern does not play very nicely with eclusions. I find it simpler to read to split the exclusions.

Reply:
Mhh it feels a bit too heavy to call `filter` just to remove `.d.ts` files if we could just add `?!(d.)` to it (previously the ignore was not correct).

That being said if you'd still prefer the filter to do the job I'd rather join them together. This way we are perfomring one extra loop just to ignore them, whic honestly feels a bit too complex.

Reply:
@karelhala I don't think glob supports ?!(d.). That is why I've gone with the additional filter.